okhttp-testing-support/src/main/java/okhttp3/PlatformRule.kt

```diff
@@ -55,55 +59,107 @@ open class PlatformRule @JvmOverloads constructor(
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_PROPERTY))
+  }
+
+  fun assumeJdk8Alpn() {
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_ALPN_PROPERTY))
+  }
+
+  fun assumeHttp2Support() {
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_PROPERTY))
   }
 
   fun assumeNotConscrypt() {
-    assumeThat(getPlatformSystemProperty(), not(CONSCRYPT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        CONSCRYPT_PROPERTY))
   }
 
   fun assumeNotJdk9() {
-    assumeThat(getPlatformSystemProperty(), not(JDK9_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK9_PROPERTY))
   }
 
   fun assumeNotJdk8() {
-    assumeThat(getPlatformSystemProperty(), not(DEFAULT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_PROPERTY))
+  }
+
+  fun assumeNotJdk8Alpn() {
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_ALPN_PROPERTY))
+  }
+
+  fun assumeNotHttp2Support() {
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_PROPERTY))
   }
 
   fun assumeJettyBootEnabled() {
     assumeTrue("ALPN Boot not enabled", isAlpnBootEnabled())
   }
 
   companion object {
-    val PROPERTY_NAME = "okhttp.platform"
-    val CONSCRYPT_PROPERTY = "conscrypt"
-    val DEFAULT_PROPERTY = "default"
-    val JDK9_PROPERTY = "jdk9"
+    const val PROPERTY_NAME = "okhttp.platform"
+    const val CONSCRYPT_PROPERTY = "conscrypt"
+    const val JDK9_PROPERTY = "jdk9"
+    const val JDK8_ALPN_PROPERTY = "jdk8alpn"
+    const val JDK8_PROPERTY = "jdk8"
 
     init {
       if (getPlatformSystemProperty() == CONSCRYPT_PROPERTY && Security.getProviders()[0].name != "Conscrypt") {
+        if (!Conscrypt.isAvailable()) {
+          System.err.println("Warning: Conscrypt not available")
+        }
+
         val provider = Conscrypt.newProviderBuilder().provideTrustManager(true).build()
         Security.insertProviderAt(provider, 1)
-      } else if (getPlatformSystemProperty() == DEFAULT_PROPERTY) {
+      } else if (getPlatformSystemProperty() == JDK8_ALPN_PROPERTY) {
         if (!isAlpnBootEnabled()) {
           System.err.println("Warning: ALPN Boot not enabled")
         }
+      } else if (getPlatformSystemProperty() == JDK8_PROPERTY) {
+        if (isAlpnBootEnabled()) {
+          System.err.println("Warning: ALPN Boot enabled unintentionally")
+        }
       }
     }
 
     @JvmStatic
-    fun getPlatformSystemProperty(): String = System.getProperty(PROPERTY_NAME, DEFAULT_PROPERTY)
+    fun getPlatformSystemProperty(): String {
+      var property: String? = System.getProperty(PROPERTY_NAME)
+
+      if (property == null) {
+        property = when (Platform.get()) {
+          is ConscryptPlatform -> CONSCRYPT_PROPERTY
+          is Jdk8WithJettyBootPlatform -> CONSCRYPT_PROPERTY
+          is Jdk9Platform -> JDK9_PROPERTY
+          else -> JDK8_PROPERTY
+        }
+      }
```

Comment:
TODO: I should have an else check that warns in the reverse case, e.g. checks a specified system property matches Platform.get().  For a later PR.